Fix docker availability handling accross platforms

breskeby opened the pull request:
This addresses some issues detecting docker availability and test skipping on windows after updating testcontainer to > 2.x. This is a fallout from our improvements on rerunning periodic builds and I ran into this repeatedly there

breskeby committed:
Fix docker availability handling accross platforms

jozala commented on test/fixtures/testcontainer-utils/src/main/java/org/elasticsearch/test/fixtures/testcontainers/DockerAvailability.java:
```diff
@@ -0,0 +1,125 @@
+        if (CI) {
+            // we dont exclude OS outside of CI environment
+            return false;
+        }
```
I may get lost a bit in the number of negations here, but isn't it the opposite to what the comment says?

From the code, I understand that if `CI == true` then we do not exclude. That means we never exclude the OS if we are in the CI, but only exclude when running locally.\
For the CI run the `isExcludedOs()` is always `false`. This value is used in the `assumeFalse` so if we are in CI we get `assumeFalse(false)`, and the test will always pass this assume invocation.

jozala reviewed (changes requested):
I see the `DockerAvailabilty` part is mostly code moved from `DockerEnvironmentAwareTestContainer`, but I've got some doubts about how this is expected to work. 
I'd like to clarify the CI comment, but I would like to also understand the general aim.

1. Do we want to ignore the problematic Docker tests runs in the CI or outside of it?
2. I understand it makes sense for the `EXCLUDED_OS` in the CI, because these OSes from `dockerOnLinuxExclusions` file cannot run Docker for whatever reason. However, I think we should fail in CI if we are on a supported OS, but Docker is unavailable (`DOCKER_PROBING_SUCCESSFUL == false`). With the current implementation, we hide the infrastructure configuration issues and silently ignore the tests whenever Docker is not available.

breskeby committed:
Fix Docker availability handling and CI logic per review

    - Invert logic in isExcludedOs to correctly apply exclusions only in CI.
    - Fail tests in CI if Docker is missing on supported OS instead of skipping.
    - Update assumption messages for clarity.

jozala reviewed (approved):
Looks good
